File: lib/models/supabase/context.py
import json
from datetime import datetime
from typing import ClassVar, Dict, Optional
from lib.models.supabase.supabase_model import SupabaseModel
from uuid import UUID
from pydantic import Field, field_validator


class ContextQuery(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "context_query"
    id: UUID
    params: Optional[Dict] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)

    @field_validator("params", mode="before")
    def validate_params(cls, v):
        if isinstance(v, str):
            return json.loads(v)
        elif isinstance(v, dict):
            return json.dumps(v)
        return v


class ContextQueryResponse(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "context_query_response"
    id: UUID
    query_id: Optional[UUID] = Field(default=None)
    response_data: Optional[Dict] = Field(default=None)
    disabled: bool = Field(default=False)
    disabled_at: Optional[datetime] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)

    @field_validator("response_data", mode="before")
    def validate_response_data(cls, v):
        if isinstance(v, str):
            return json.loads(v)
        elif isinstance(v, dict):
            return json.dumps(v)
        return v


class ContextQueryResult(SupabaseModel):
    TABLE_NAME: ClassVar[str] = "context_query_result"
    id: UUID
    query_id: Optional[UUID] = Field(default=None)
    result_data: Optional[Dict] = Field(default=None)
    created_at: Optional[datetime] = Field(default=None)
    updated_at: Optional[datetime] = Field(default=None)
    owner_id: Optional[UUID] = Field(default=None)
    organization_id: Optional[UUID] = Field(default=None)

    @field_validator("result_data", mode="before")
    def validate_result_data(cls, v):
        if isinstance(v, str):
            return json.loads(v)
        elif isinstance(v, dict):
            return json.dumps(v)
        return v

File: lib_js/models/supabase/context.js
import { SupabaseModel } from "./supabaseModel.js";

// Define ContextQuery Model
export class ContextQuery extends SupabaseModel {
    static TABLE_NAME = "context_query";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: true, dbColumn: "id" },
        params: { type: "json", required: false, dbColumn: "params" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define ContextQueryResponse Model
export class ContextQueryResponse extends SupabaseModel {
    static TABLE_NAME = "context_query_response";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: true, dbColumn: "id" },
        queryId: { type: "uuid", required: false, dbColumn: "query_id" },
        responseData: {
            type: "json",
            required: false,
            dbColumn: "response_data"
        },
        disabled: { type: "boolean", required: false, dbColumn: "disabled" },
        disabledAt: { type: "date", required: false, dbColumn: "disabled_at" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}

// Define ContextQueryResult Model
export class ContextQueryResult extends SupabaseModel {
    static TABLE_NAME = "context_query_result";
    static TABLE_FIELDS = {
        id: { type: "uuid", required: true, dbColumn: "id" },
        queryId: { type: "uuid", required: false, dbColumn: "query_id" },
        resultData: { type: "json", required: false, dbColumn: "result_data" },
        createdAt: { type: "date", required: false, dbColumn: "created_at" },
        updatedAt: { type: "date", required: false, dbColumn: "updated_at" },
        ownerId: { type: "uuid", required: false, dbColumn: "owner_id" },
        organizationId: {
            type: "uuid",
            required: false,
            dbColumn: "organization_id"
        }
    };
}
